test/alternator/test_batch.py

```diff
@@ -402,4 +402,22 @@ def test_batch_get_item_partial(scylla_only, dynamodb, test_table_sn):
         responses.extend(reply['Responses'][test_table_sn.name])
     assert multiset(responses) == multiset(
         [{'p': p, 'c': i, 'content': content} for i in range(count)])
-    assert some_keys_were_unprocessed
\ No newline at end of file
+    assert some_keys_were_unprocessed
+
+# Test that if the batch read failure is total, i.e. all read requests
+# failed, it's reported as an error and not as a regular response with
+# UnprocessedKeys set to all given keys.
```

Comment:
This makes me wonder what leads us to believe that this **needs to be** the behavior. I mean, if we do a 3-read batch and two of the reads failed, we get one type of error, but if all 3 failed, we get a different error. Why is this important? Do we have a reason to believe that DynamoDB behaves this way?